File: src/test/java/br/com/controle/financeiro/controller/api/linkbuilder/InstitutionDTOResourceAssemblerTest.java
package br.com.controle.financeiro.controller.api.linkbuilder;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertTrue;

import java.util.UUID;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.InjectMocks;
import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.hateoas.EntityModel;
import org.springframework.test.context.ActiveProfiles;
import org.springframework.test.context.junit4.SpringRunner;

import br.com.controle.financeiro.model.dto.InstitutionDTO;

@RunWith(SpringRunner.class)
@SpringBootTest(classes = { InstitutionDTOResourceAssembler.class })
@AutoConfigureMockMvc
@ActiveProfiles(profiles = "test")
public class InstitutionDTOResourceAssemblerTest {

	@InjectMocks
	private InstitutionDTOResourceAssembler resourceAssembler;


	@Test
	public void testToResource() {
		InstitutionDTO institutionMock = new InstitutionDTO();
		institutionMock.setId(UUID.randomUUID());
		institutionMock.setName("mock");
		EntityModel<InstitutionDTO> response = resourceAssembler.toModel(institutionMock);
		assertTrue(response.hasLinks());
		assertNotNull(response.getLink("self"));
		assertNotNull(response.getLink("institutions"));
		assertEquals(institutionMock, response.getContent());
	}

}

File: src/test/kotlin/br/com/controle/financeiro/controller/api/linkbuilder/InstitutionDTOResourceAssemblerTest.kt
package br.com.controle.financeiro.controller.api.linkbuilder

import org.junit.jupiter.api.Test
import org.junit.jupiter.api.Assertions
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.ActiveProfiles
import org.springframework.beans.factory.annotation.Autowired
import br.com.controle.financeiro.controller.api.ins
import br.com.controle.financeiro.model.dto.InstitutionDTO

@SpringBootTest
@ActiveProfiles(profiles = ["test"])
class InstitutionDTOResourceAssemblerTest (
    @Autowired private val resourceAssembler: InstitutionDTOResourceAssembler
 ) {   @Test
    fun testToResource() {
        val institutionMock = InstitutionDTO(ins.name, ins.identifier, ins.id)
        val response = resourceAssembler.toModel(institutionMock)
        Assertions.assertTrue(response.hasLinks())
        Assertions.assertNotNull(response.getLink("self"))
        Assertions.assertNotNull(response.getLink("institutions"))
        Assertions.assertEquals(institutionMock, response.content)
    }
}
